src/test/java/TaskToEventTest.java

```diff
@@ -0,0 +1,157 @@
+import data_gateway.CalendarManager;
+import data_gateway.EventReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import services.event_creation.CalendarEventModel;
+import services.event_from_task_creation.EventScheduler;
+import services.event_from_task_creation.TaskToEvent;
+import services.task_presentation.TaskInfo;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TaskToEventTest {
+    TaskToEvent taskToEvent;
+    TaskInfo taskInfo;
+    EventScheduler scheduler;
+    CalendarManager manager;
+
+    @BeforeEach
+    void setup() {
+        taskToEvent = new TaskToEvent();
+        taskInfo = new MockTaskInfo();
+        manager = new MockCalendarManager();
+        scheduler = new EventScheduler(manager);
+    }
+
+    @Test
+    void getAvailableTime() {
+        LocalDateTime expected = scheduler.getAvailableTime(
+                new ArrayList<>() , taskInfo.getDuration()).truncatedTo(ChronoUnit.MINUTES);
+        LocalDateTime actual = taskToEvent.getAvailableTime(taskInfo, scheduler, new ArrayList<>());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void checkTimeAvailability() {
+        LocalDateTime userSuggestedTime = LocalDateTime.of(2021, 11, 26, 10, 0);
+        boolean expected = scheduler.isAvailable(userSuggestedTime.toLocalTime(), taskInfo.getDuration(),
```

Comment:
You should not be calling the method for expected value. It should a hard coded.

Reply:
Calling EventScheduler is the expected behaviour of this method (see Line 19 and Line 31 of TestToEvent)